README.md

```diff
@@ -31,37 +36,255 @@ npm install @uploadcare/upload-client --save
+
+  group(files: Uuid[], settings: SettingsInterface): CancelableThenableInterface<GroupInfoInterface>
+
+  groupInfo(id: GroupId, settings: SettingsInterface): CancelableThenableInterface<GroupInfoInterface>
+
+  multipartStart(file: FileData, settings: SettingsInterface): CancelableThenableInterface<MultipartStartResponse>
+
+  multipartUpload(file: FileData, parts: MultipartPart[], settings: SettingsInterface): BaseThenableInterface<any>
+
+  multipartComplete(uuid: Uuid, settings: SettingsInterface): CancelableThenableInterface<FileInfoInterface>
+}
+```
+
+### Low Level API
+
+If you want to use our low level API, you can do that something like this:
+
+```javascript
+import UploadClient from '@uploadcare/upload-client'
+
+const client = new UploadClient({publicKey: 'YOUR_PUBLIC_KEY'})
+
+client.api.request({path: 'info', query})
+  .then(response => console.log(response.data))
+```
+
+### Settings
+
+#### publicKey: string
+
+The main use of a `publicKey` is to identify a target project for your uploads. 
+It is required when using Upload API.
+
+#### baseCDN: string
+
+Defines your schema and CDN domain. Can be changed to one of 
+the predefined values or your custom CNAME. 
+
+Defaults to `https://ucarecdn.com/`.
+
+#### baseURL: string
+
+API base URL. 
+
+Defaults to `https://upload.uploadcare.com`
+
+#### fileName: string
+
+You can specify an original filename. 
+
+Defaults to `original`.
+
+#### doNotStore: boolean
+
+Forces files uploaded with a widget not to be stored. 
+For instance, you might want to turn this on when automatic file storing 
+is enabled in your project, but you do not want to store files uploaded 
+with a particular instance.
+
+#### secureSignature: string
+
+In case you enable signed uploads for your project, you’d need to provide 
+the client with signature and expire. 
+
+The `secureSignature` is an MD5 hex-encoded hash from a concatenation 
+of `API secret key` and `secureExpire`.
+
+#### secureExpire: string
+
+Stands for the Unix time to which the signature is valid, e.g., `1454902434`.
+
+#### integration: string
+
+`X-UC-User-Agent` header value.
+
+Defaults to `UploadcareUploadClient/${version}${publicKey} (JavaScript${integration})`
+
+#### checkForUrlDuplicates: boolean
+
+Runs the duplicate check and provides the immediate-download behavior.
+
+#### saveUrlForRecurrentUploads: boolean
+
+Provides the save/update URL behavior. The parameter can be used 
+if you believe a `sourceUrl` will be used more than once. 
+Using the parameter also updates an existing reference with a new 
+`sourceUrl` content.
+
+#### source: string
+
+Upload source. Each upload method has its own (as example: `local`, `url`, etc.)
+
+#### jsonpCallback: string
+
+Sets the name of your JSONP callback function to create files group from 
+a set of files by using their UUIDs.
+
+#### pollingTimeoutMilliseconds: number
+
+Internally upload client using polling to ensure that file is available on CDN 
+or its uploaded from url.
```

Comment:
Internally, Upload Client implements polling to ensure that a file is available on CDN or has finished uploading from URL.